tests/testutils/telemetry/logs.go

```diff
@@ -286,30 +281,55 @@ func (resourceLogs ResourceLogs) ContainsAll(contains ResourceLogs) (bool, error
 							}
 							if len(missingLogs) != 0 {
 								return false, fmt.Errorf(
-									"%v doesn't contain all of %v.  Missing Logs: %s",
-									ilm.Logs, expectedILM.Logs, missingLogs)
+									"%v doesn't contain all of %v. Missing Logs: %s",
+									ilm.Logs, expectedSL.Logs, missingLogs)
 							}
 						}
 					}
-					if !InstrumentationScopeMatched {
-						missingInstrumentationLibraries = append(missingInstrumentationLibraries, expectedILM.Scope.String())
+					if matchingInstrumentationScope != "" {
+						// no longer globally missing
+						delete(missingInstrumentationScopes, matchingInstrumentationScope)
+					} else {
+						innerMissingInstrumentationScopes[expectedSL.Scope.String()] = struct{}{}
 					}
 				}
-				if len(missingInstrumentationLibraries) != 0 {
-					return false, fmt.Errorf(
-						"%v doesn't contain all of  %v.  Missing InstrumentationLibraries: %s",
-						resourceLog.ScopeLogs, expectedResourceLog.ScopeLogs, missingInstrumentationLibraries)
+				if len(innerMissingInstrumentationScopes) != 0 {
```

Comment:
I'm really not happy w/ how these are progressing and plan on making a more generic central helper instead of having to repeat the logic for each telemetry source. It didn't seem clear to me how to best do this in this go so prefer to punt it for now.